src/worker.rs

```diff
@@ -141,6 +143,62 @@ impl Worker {
         })
     }
 
+    // Returns a validation response with policy-server specific
+    // constraints taken into account:
+    // - A policy might have tried to mutate while the policy-server
+    //   configuration does not allow it to mutate
+    // - A policy might be running in "Monitor" mode, that always
+    //   accepts the request (without mutation), logging the answer
+    fn validation_response_with_constraints(
+        policy_id: &str,
+        policy_mode: &PolicyMode,
+        allowed_to_mutate: bool,
+        validation_response: ValidationResponse,
+    ) -> ValidationResponse {
+        if validation_response.patch.is_some() && !allowed_to_mutate {
```

Comment:
Shouldn't we look at `policy_mode` earlier inside of this method?

Let's assume I declare a policy with the following attributes:
  * allowed to mutate: false
  * policy mode: monitor

Let's assume the evaluation of this policy would lead to a mutation.

This would lead Policy Server to reply with a rejection message. What I expect is to have the policy accept the object, with no mutation done (because `allowed_to_mutate` is `false`)

Reply:
I don't think this would be a misconfiguration. The way I see it, a user who takes a mutating policy and marks that as validating only is doing that on purpose.
That's why I say we should still accept the request

Reply:
Let me circle back a bit then. What you specify here:

"What I expect is to have the policy accept the object, with no mutation done (because allowed_to_mutate is false)"

is not what is going to happen. In the case you have described, the policy server will reject the request with an error message, saying the policy tried to mutate a request while it was configured as validating only.

Reply:
Oh yes, that makes sense! Sorry about the noise and thanks for the explanation